src/RequestHandler.php

```diff
@@ -292,6 +292,8 @@ public function slaveClosed()
         $this->incoming->end();
         if ($this->maxExecutionTime > 0) {
             $this->loop->cancelTimer($this->maxExecutionTimer);
+            //Explicitly null the property to avoid a cyclic memory reference
+            $this->maxExecutionTimer = null;
```

Comment:
This is one of the things we've been doing in ReactPHP packages as well: clean things up as soon as we don't need them any more. Literally setting them to `null` like this, it might look to simple but it actually helps you out. Another thing is that with `gc_collect_cycles()` you can measure if there is still something in there you might have been able to clean up before the gc catches it, increasing performance even more.